domain/containerimageresourcestore/service/service.go

```diff
@@ -121,3 +125,31 @@ func (s Service) Remove(
 	}
 	return nil
 }
+
+// getSizeAndFingerprint gets the size and fingerprint that a DockerImageDetails
+// struct will have when exported as a blob from the container image metadata
+// service.
+func getSizeAndFingerprint(dockerDetails docker.DockerImageDetails) (int64, store.Fingerprint, error) {
+	data, err := json.Marshal(dockerDetails)
```

Comment:
We shouldn't be marshalling a `DockerImageDetails` ourselves... this isn't ours and a library update could invalidate **all** our fingerprints. 

Instead, we should marshall something that we own.

This will happen, it's only a matter of time.